### PR TITLE
fix(loss): avoid pkg_resources in linear CE

### PR DESCRIPTION
**Pull request by akoumpa**
# What does this PR do ?

Avoids the setuptools `pkg_resources` deprecation warning from `linear_ce.py` when fused linear CE recipes patch cut-cross-entropy Triton version checks.

# Changelog

- Replace `pkg_resources` distribution/version lookup with `importlib.metadata` and `packaging.version`.
- Fall back from `pytorch-triton` package metadata to `triton` when checking the installed Triton version.
- Update linear CE unit tests for the new metadata path.

# Before your PR is "Ready for review"

**Pre checks**:

- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA-NeMo/Automodel/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests?
- [x] Did you add or update any necessary documentation? Not needed; code/test only.

# Additional Information

Verified with:

- `python -m pytest tests/unit_tests/loss/test_linear_ce.py -q`
- `python -m ruff check --fix nemo_automodel/components/loss/linear_ce.py tests/unit_tests/loss/test_linear_ce.py`
- `python -W error::UserWarning - <<'PY' ...`

### CODE REVIEW

**Comment by akoumpa** [comment]
i want to avoid warning spamming in 26.06 so i'm including this as well

**Review by yuhezhang-ai — APPROVED** [review]
